### PR TITLE
Running only Solid-OIDC provider without storage (with existing WebIDs only!)

### PR DESCRIPTION
**Pull request by elf-pavlik**
It's a very early draft, this draft PR serves as an invitation to offer feedback and possibly collaborate.

### CODE REVIEW

File: solo-oidc/config-solo-oidc.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    "css:config/app/variables/default.json",
+    "css:config/http/handler/default.json",
+    "css:config/http/middleware/default.json",
+    "css:config/http/notifications/disabled.json",
+    "css:config/http/server-factory/http.json",
+    "css:config/http/static/default.json",
+    "css:config/identity/access/public.json",
+    "css:config/identity/email/default.json",
+    "css:config/identity/handler/default.json",
+    "css:config/identity/ownership/unsafe-no-check.json",
```
Comment (elf-pavlik) [inline solo-oidc/config-solo-oidc.json:16]:
```suggestion
   "css:config/identity/ownership/token.json",
```

TODO before marking ready for review

**Comment by elf-pavlik** [comment]
I improvised with the way I'm overwriting the HTML templates. The way I do it doesn't seem to fully work, for example, I can't use variables like `isBlankForm ` or `prefilled` so I just removed them for now not to get errors. I will probably also need to replace some inputs with hidden inputs if my overwrite doesn't allow selecting any other option.

**Comment by joachimvh** [comment]
I'm not sure if putting everything needed for this in recipes is the solution. For example, for the config part of only having OIDC, which comes down to disabling access to the LDP part, it might be more interesting to just have another import option for the `config/http/handler` import in the CSS repository itself..

The main goal of this repository was also to showcase how the configurations could be combined, so-called recipes, so adding custom HTML templates here might be too much.

> I can't use variables like `isBlankForm ` or `prefilled`

Having a look at the code, I think `prefilled` might have been removed from the code at some point but never updated in the templates as the fact that it disappeared didn't actually impact the result.

**Comment by elf-pavlik** [comment]
This PR is mostly learning by doing for me, plus I saw on the chat that someone else also needs something similar so doing it in the open might be a good idea.

Would you be able to start a branch in the CSS repo with an initial example of another import option you suggested?
I  haven't used EJS 

When it comes to templates, do you think it would be possible that based on configuration, only some parts of the specific page would be rendered? I pretty much only deleted from them any functionality which I don't want to enable, doing it via config would be much cleaner.

The current state of modified `/idp/register/` 

image

I still would like to remove the radio button and the checkbox, only leaving input for existing WebID and the account credentials.

**Comment by joachimvh** [comment]
> Would you be able to start a branch in the CSS repo with an initial example of another import option you suggested?

I can have a look at it later, but it would be another variant of this import: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/main/config/http/handler/default.json,
but then with the necessary lines removed.



> When it comes to templates, do you think it would be possible that based on configuration, only some parts of the specific page would be rendered?

The only way to do this is to pass extra parameters to the templates, so these would have to be included in the configuration as well. This is done here for example: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/fbe60371b9643961200c6b7aff736c6269a8331f/config/identity/registration/enabled.json#L44
But this can get ugly quite fast if you are dropping template-specific variables all over the place. The one linked above is probably going to disappear in the future specifically for that reason. It would probably be better to have specific templates for different configurations.

**Comment by elf-pavlik** [comment]
> But this can get ugly quite fast if you are dropping template-specific variables all over the place.

I agree, also templates using those variables would end up with a lot of conditionals, especially showing radio buttons or not could get pretty convoluted.

>  It would probably be better to have specific templates for different configurations.

I guess all of those templates could still use a common set of reusable partials/components. This way each template would only include what I need to show.

-----

I'm not sure if I'm specifying which template should be used in the correct way. I ended up copying into my config the whole content of `config/identity/registration/enabled.json` just to set `"HtmlViewHandler:_templates_key"` to a different value. TBH I hope there is an easier way to accomplish it.

**Comment by joachimvh** [comment]
> TBH I hope there is an easier way to accomplish it.

You can use Components.js overrides. See https://componentsjs.readthedocs.io/en/latest/configuration/configurations/overrides/ and/or https://github.com/CommunitySolidServer/tutorials/blob/main/custom-configurations.md#overriding-preset-values

**Comment by elf-pavlik** [comment]
```jsonld
"@graph": [
    {
      "@type": "Override",
      "overrideInstance": { "@id": "urn:solid-server:auth:password:HtmlViewHandler" },
      "overrideParameters": {
        "templates": [
          {
            "HtmlViewHandler:_templates_key": "./templates/register.html.ejs",
            "HtmlViewHandler:_templates_value": { "@id": "urn:solid-server:auth:password:RegistrationRoute" }
          }
        ]
      }
    }
  ]
```

results in error

```
Error while parsing file "/home/elf-pavlik/code/solid/CSS-Recipes/solo-oidc/config-solo-oidc.json": Invalid predicate IRI: templates
```

When I look at [Component.js test for Override](https://github.com/LinkedSoftwareDependencies/Components.js/blob/b15d640188a47246b7923d9018bd5fc46bf7fd04/test/assets/config-override-reverse.jsonld#L16) it uses prefixed predicates, which works fine.

Context generated from CSS uses [Scoped Context](https://www.w3.org/TR/json-ld11/#scoped-contexts)

```jsonld
"HtmlViewHandler": {
        "@id": "css:dist/identity/interaction/HtmlViewHandler.jsonld#HtmlViewHandler",
        "@prefix": true,
        "@context": {
          "templateEngine": {
            "@id": "css:dist/identity/interaction/HtmlViewHandler.jsonld#HtmlViewHandler_templateEngine"
          },
          "index": {
            "@id": "css:dist/identity/interaction/HtmlViewHandler.jsonld#HtmlViewHandler_index"
          },
          "templates": {
            "@id": "css:dist/identity/interaction/HtmlViewHandler.jsonld#HtmlViewHandler_templates",
            "@container": "@list"
          },
          "gine": {
            "@id": "css:dist/identity/interaction/HtmlViewHandler.jsonld#HtmlViewHandler_templateEngine"
          }
        }
      },
```

**Comment by joachimvh** [comment]
I've added some config options in https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1739 which should cover relevant cases.

**Pull request closed without merging**